src/libraries/System.Private.CoreLib/src/System/IO/TextReader.cs

```diff
@@ -342,15 +342,41 @@ private sealed class NullTextReader : TextReader
         {
```

Comment:
I wonder if we even need NullTextReader?  Seems like `public static readonly TextReader Null = new NullTextReader();` could be changed to instead construct a `NullStreamReader`.  It's possible there's some reason I'm missing not to do that, but it'd be nice to avoid the duplication if possible. @adamsitnik, thoughts?

Reply:
I wondered this, too, as I refactored the classes to be almost identical; the only thing I can think of changing/breaking is code that (essentially) performs `TextReader.Null is TextReader and not StreamReader` (or similar `switch`/`case` or `if`/`else`).

If setting `TextReader.Null` to a `NullStreamReader` (and deleting `NullTextReader`) is desirable, let me know and I'm happy to make the change.

Reply:
The scenario provided by @bgrainger is valid, but I hope that nobody does that.

I've taken a look at how these two properties are [typically used](https://cs.github.com/?q=StreamReader.Null%20language%3AC%23) and I think that it's relatively safe to reduce the code duplication and remove `NullTextReader`.